### ISSUE TITLE
The source code in the Dependencies and window chapters is outdated.

### ISSUE THREAD

**Issue by whtsht** [sotrh/learn-wgpu#546]
Hello, I'm enjoying reading learn-wgpu.

As a result of changes made in https://github.com/rust-windowing/winit/pull/3056, the following adjustments are necessary:

- Replace `control_flow = ControlFlow::Exit` with `control_flow.exit()`.
- Change `event_loop.run(move |event, _, control_flow| { ... })` to `event_loop.run(move |event, control_flow| { ... })`.

The source code on [github](https://github.com/sotrh/learn-wgpu/blob/master/code/beginner/tutorial1-window/src/lib.rs) has been updated. However, it appears that the code on the [article](https://sotrh.github.io/learn-wgpu/beginner/tutorial1-window/#the-code) remains unchanged.

Readers who copy and paste this code will encounter compilation errors. This could discourage beginners from continuing with the article.

I can submit a pull request to address this issue.

**Comment by aerosayan** [comment]
Exactly the problem I faced. 

I'm new to rust, and started learning wpgu two days ago.

I was irritated to find that this tutorial recommended by wgpu was not even compiling, and the new API is so complex and unstable that I couldn't use it.

Only after one day of trying different things, I found out that this tutorial series had a github repo, and the code in the repo, was working correctly. 

I see three critical reason that caused this confusion:

1. The article contains bugs, and even the first tutorial doesn't compile. This will prevent first time visitors from continuing  the series.
2. The over-dependence on the "web" aspect of wgpu is not useful for me. I want to develop a desktop application, and millions of other developers would want to do the same. Web dev is not a priority for us. Perhaps it would be better if the tutorial series was split into two parts. One for desktop development, and one for web development. 
3. The article's UI design makes it difficult to find the link to this repo. As seen in this picture, the link to this repo is given at the bottom, in small text, and is almost invisible to a first time visitor. I did not notice it for one day. The source code should be linked at the top banner, or spread out multiple times throughout the article. The link to this repo should be included at the top of the left pane. Github icons should be used at the top of the repo (or anywhere you deem useful) to link to this repo.

![1](https://github.com/sotrh/learn-wgpu/assets/39849461/dd78bd7f-2e83-418b-85e9-575d1ee46d5f)
